File: TelefQuito/02.CLIWEB/TELFQUITO_RESTFUL_JAVA_WEB/services/TelefonoService.js
const axios = require("axios");

class TelefonoService {
  constructor() {
    this.baseUrl = "http://localhost:8080/TELFQUITO_RESTFUL_JAVA/resources";
  }

  async getTelefonoById(id) {
    try {
      const response = await axios.get(`${this.baseUrl}/wstelefonos/${id}`);
      return response.data.telefono;
    } catch (error) {
      console.error("Get Telefono By Id Error:", error);
      throw error;
    }
  }

  async getAllTelefonos() {
    try {
      const response = await axios.get(`${this.baseUrl}/wstelefonos`);
      return response.data;
    } catch (error) {
      console.error("Get All Telefonos Error:", error);
      throw error;
    }
  }

  async insertTelefono(telefono) {
    try {
      const response = await axios.post(`${this.baseUrl}/telefonos`, telefono);
      return response.data.result;
    } catch (error) {
      console.error("Insert Telefono Error:", error);
      throw error;
    }
  }

  async updateTelefono(telefono) {
    try {
      const response = await axios.put(
        `${this.baseUrl}/telefonos/${telefono.id}`,
        telefono
      );
      return response.data.result;
    } catch (error) {
      console.error("Update Telefono Error:", error);
      throw error;
    }
  }
}

module.exports = new TelefonoService();

File: TelefQuito/02.CLIMOV/app/src/main/java/com/example/telfquito_restful_java/service/TelefonoService.java
package com.example.telfquito_restful_java.service;

import com.example.telfquito_restful_java.models.TelefonoModel;

import java.util.List;

import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;

public interface TelefonoService {

    // Get all telefonos
    @GET("wstelefonos")
    Call<List<TelefonoModel>> getAllTelefonos();

    // Get a telefono by ID
    @GET("wstelefonos/{id}")
    Call<TelefonoModel> getTelefonoById(@Path("id") int id);

    // Insert a new telefono
    @POST("wstelefonos")
    Call<ResponseBody> insertTelefono(@Body TelefonoModel telefono);

    // Update an existing telefono (Returns plain text)
    @PUT("wstelefonos")
    Call<ResponseBody> updateTelefono(@Body TelefonoModel telefono);
}
